pkg/cmd/subcommand/demo.go

```diff
@@ -0,0 +1,490 @@
+
+	fmt.Fprintln(w, ti.BeforeTask("Launching OpenFunction demo...\n"+
+		"The following components will be installed for this demo:"))
+	printInventory(inventoryPending)
+
+	if i.DryRun {
+		return nil
+	}
+	if !Continue() {
+		return nil
+	}
+
+	grp1, g1ctx := errgroup.WithContext(ctx)
+
+	start := time.Now()
+
+	grp1.Go(func() error {
+		return i.InstallKind(g1ctx, operator)
+	})
+
+	if err := grp1.Wait(); err != nil {
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	_, cl, err := client.NewKubeConfigClient()
+	if err != nil {
+		return err
+	}
+	installinventoryPending, err := inventory.GetInventory(
+		cl,
+		i.RegionCN,
+		true,
+		true,
+		true,
+		true,
+		true,
+		false,
+		i.OpenFunctionVersion,
+	)
+	if err != nil {
+		return errors.Wrap(err, "failed to get pending inventory")
+	}
+	operator.Inventory = installinventoryPending
+
+	fmt.Fprintln(w, ti.BeforeTask("Start installing OpenFunction and its dependencies.\n"+
+		"Here are the components and corresponding versions to be installed:"))
+	printInventory(inventory.GetVersionMap(installinventoryPending))
+
+	grp2, g2ctx := errgroup.WithContext(ctx)
+	grp2.Go(func() error {
+		return i.InstallDapr(g2ctx, operator)
+	})
+	grp2.Go(func() error {
+		return i.installKeda(g2ctx, cl, operator)
+	})
+	grp2.Go(func() error {
+		return i.installKnativeServing(g2ctx, cl, operator)
+	})
+	grp2.Go(func() error {
+		return i.installShipwright(g2ctx, cl, operator)
+	})
+	grp2.Go(func() error {
+		return i.installCertManager(g2ctx, cl, operator)
+	})
+
+	if err := grp2.Wait(); err != nil {
+		util.CheckErr(i.DeleteKind())
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	grp3, g3ctx := errgroup.WithContext(ctx)
+
+	grp3.Go(func() error {
+		return i.installOpenFunction(g3ctx, cl, operator)
+	})
+	if err := grp3.Wait(); err != nil {
+		i.DeleteKind()
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	grp4, g4ctx := errgroup.WithContext(ctx)
+
+	grp4.Go(func() error {
+		if err := i.RunOpenFunctionDemo(g4ctx, cl, operator); err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err := grp4.Wait(); err != nil {
+		util.CheckErr(i.DeleteKind())
+		return errors.New(ti.TaskFail(err.Error()))
+	}
+
+	end := time.Since(start)
+	fmt.Fprintln(w, ti.AllDone(end))
+	return nil
+}
+
+func (i *Demo) checkConditionsAndGetInventory() map[string]string {
+	i.OpenFunctionVersion = "v0.4.0"
+	inventory := map[string]string{
+		"OpenFunction": i.OpenFunctionVersion,
+	}
+
+	return inventory
+}
+
+func (i *Demo) InstallKind(ctx context.Context, operator *common.Operator) error {
+	ctx, done := context.WithCancel(ctx)
+	defer done()
+	ti := util.NewTaskInformer("Kind")
+
+	fmt.Fprintln(w, ti.TaskInfo("Installing Kind..."))
+	fmt.Fprintln(w, ti.TaskInfo("Downloading Kind binary..."))
+	if err := operator.DownloadKind(ctx); err != nil {
+		return errors.Wrap(err, ti.TaskFailWithTitle("Failed to download Kind"))
+	}
+	fmt.Fprintln(w, ti.TaskInfo("Initializing Kind with Kubernetes mode..."))
```

Comment:
"Initializing Kind with Kubernetes mode..." -> "Creating cluster..." or something like this